src/solver/glu/backward_glu.cpp

```diff
@@ -0,0 +1,118 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <miopen/kernel_info.hpp>
+#include <miopen/mlo_internal.hpp>
+#include <cstddef>
+#include <miopen/datatype.hpp>
+#include <miopen/kernel_build_params.hpp>
+#include <miopen/glu/invoke_params.hpp>
+#include <miopen/glu/solvers.hpp>
+#include <miopen/glu.hpp>
+#include <miopen/target_properties.hpp>
+
+#define LOCAL_SIZE 256
+
+namespace miopen {
+
+namespace solver {
+
+namespace glu {
+
+bool GLUBackward::IsApplicable(const ExecutionContext& context,
+                               const miopen::glu::ProblemDescription& problem) const
```

Comment:
I have implemented datatype check at `problem_description.hpp`

Reply:
I meant that you should add some code here that checks whether certain data types are supported or not

Reply:
I added check for 3 support data types: `float`, `fp16` and `bfp16`

Reply:
The point is that there can be several solvers for the same algorithm, one can support all data types, and another only one. What happens if we add a new data type that is not supported by any solver? If the check is performed in the solver, it will be necessary to make edits only in it, or add a new solver.